### PR TITLE
NOISSUE - Fix default nats pubsub subject

### PR DESCRIPTION
**Pull request by manuio**
Signed-off-by: Manuel Imperiale <manuel.imperiale@gmail.com>

### CODE REVIEW

**Review by blokovi — DISMISSED** [review]

**Review by mteodor — DISMISSED** [review]

**Review by drasko — CHANGES_REQUESTED** [review]
I agree that sending `topic` when we already have `msg.channel` in `msg` is redundant, but @dusanb94 had a rationale in developing the interface and he must explain.

**Review by mteodor — DISMISSED** [review]

File: transformers/senml/transformer.go [hunk 1]
```diff
@@ -59,9 +57,10 @@ func (t transformer) Transform(msg messaging.Message) (interface{}, error) {
 	msgs := make([]Message, len(normalized.Records))
 	for i, v := range normalized.Records {
 		// Use reception timestamp if SenML messsage Time is missing
-		t := v.Time
-		if t == 0 {
-			t = float64(time.Now().UnixNano())
+		time := v.Time
```
Comment (drasko) [inline transformers/senml/transformer.go]: I would avoid the name `time`, as it relates to Go library name... I prefer `t` in this case.

File: transformers/senml/transformer.go [hunk 2]
```diff
@@ -59,9 +57,10 @@ func (t transformer) Transform(msg messaging.Message) (interface{}, error) {
...
+		if time == 0 {
+			// Convert the timestamp into float64 with nanoseconds precision
+			time = float64(msg.Created.GetSeconds()) + float64(msg.Created.GetNanos())/float64(1e9)
```
Comment (drasko) [inline transformers/senml/transformer.go]: This looks like precision in seconds to me. If you wanted nanoseconds, you should probably do something like:
```go
float64(msg.Created.GetSeconds() * 1e9 + msg.Created.GetNanos())
```

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by dborovcanin — DISMISSED** [review]

**Review by mteodor — DISMISSED** [review]

**Comment by dborovcanin** [comment]
> I agree that sending `topic` when we already have `msg.channel` in `msg` is redundant, but @dusanb94 had a rationale in developing the interface and he must explain.

I decided to switch to topic-based pub-sub so that we can have a clean `Pub` and `Sub` interfaces. Also, we don't have to necessarily publish to the `msg.Channel` in the Publisher implementation, but rather let the one who uses publisher decide where to publish. This way, we keep specification (interface), implementation (NATS Pub and PubSub), and usage clean (the one who uses publisher chooses where to publish). Also, we should avoid modifying topic in the implementation, because of possible problems such as the one this PR resolves. We still have it in the `Publish` method implementation (adding channels prefix), though.

**Review by dborovcanin — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**